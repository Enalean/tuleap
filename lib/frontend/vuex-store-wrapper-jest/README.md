# @tuleap/vuex-store-wrapper-jest

Provides a helper to help test Vuex stores in Jest tests.

This package exists for legacy reasons. New apps requiring complex state management should prefer
[Pinia](https://pinia.vuejs.org/) over Vuex. See the [ADR on this topic](../../../docs/decisions/0011-js-framework.md).
