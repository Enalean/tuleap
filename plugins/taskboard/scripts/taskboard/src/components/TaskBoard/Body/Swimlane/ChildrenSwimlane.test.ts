/*
 * Copyright (c) Enalean, 2019 - present. All Rights Reserved.
 *
 * This file is a part of Tuleap.
 *
 * Tuleap is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Tuleap is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Tuleap. If not, see <http://www.gnu.org/licenses/>.
 */

import type { Wrapper } from "@vue/test-utils";
import { shallowMount } from "@vue/test-utils";
import ChildrenSwimlane from "./ChildrenSwimlane.vue";
import { createStoreMock } from "@tuleap/vuex-store-wrapper-jest";
import type { Card, ColumnDefinition, Swimlane } from "../../../../type";
import type { RootState } from "../../../../store/type";

function createWrapper(swimlane: Swimlane): Wrapper<Vue> {
    return shallowMount(ChildrenSwimlane, {
        mocks: {
            $store: createStoreMock({
                state: {
                    column: {
                        columns: [
                            {
                                id: 2,
                                label: "To do",
                                mappings: [{ tracker_id: 7, accepts: [{ id: 49 }] }],
                            } as ColumnDefinition,
                            {
                                id: 3,
                                label: "Done",
                                mappings: [{ tracker_id: 7, accepts: [{ id: 50 }] }],
                            } as ColumnDefinition,
                        ],
                    },
                } as RootState,
            }),
        },
        propsData: { swimlane },
    });
}

describe("ChildrenSwimlane", () => {
    it(`displays the parent card in its own cell and children in cells`, () => {
        const swimlane: Swimlane = {
            card: { id: 43 } as Card,
            children_cards: [],
            is_loading_children_cards: true,
        };
        const wrapper = createWrapper(swimlane);

        expect(wrapper.element).toMatchSnapshot();
    });
});
