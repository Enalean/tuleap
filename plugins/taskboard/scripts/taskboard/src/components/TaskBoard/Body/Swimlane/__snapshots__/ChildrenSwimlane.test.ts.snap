// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChildrenSwimlane displays the parent card in its own cell and children in cells 1`] = `
<div
  class="taskboard-swimlane"
  data-navigation="swimlane"
  tabindex="0"
>
  <parent-cell-stub
    swimlane="[object Object]"
  />
   
  <drop-container-cell-stub
    column="[object Object]"
    swimlane="[object Object]"
  />
  <drop-container-cell-stub
    column="[object Object]"
    swimlane="[object Object]"
  />
</div>
`;
